src/AddressSpace.cc

```diff
@@ -1040,10 +1040,16 @@ void AddressSpace::fix_stack_segment_start(const Mapping& mapping,
 }
 
 Mapping AddressSpace::vdso() const {
+  return vdso_pair().first;
+}
+
+MappingResourcePair AddressSpace::vdso_pair() const{
   assert(!vdso_start_addr.is_null());
-  return mapping_of(vdso_start_addr).first;
+  auto mapping = mapping_of(vdso_start_addr);
+  return mapping;
 }
```

Comment:
I think it would be simpler to eliminate `vdso_pair()` and just have `vdso()` check the pseudodevice and return an empty `Mapping()` if there's no mapping or it's the wrong pseudodevice.